### ISSUE TITLE
Effects of input before startup not displayed on macOS

### ISSUE THREAD

**Issue by mtoohey31** [helix-editor/helix#4668]
### Summary

On macOS, if you enter input before helix has started, the effects of that input are not displayed until you enter another keystroke after startup.

### Reproduction Steps

 

I tried this:

1. `sleep 5 && hx`
2. While sleep is still running and helix has not started, input characters that will do something in helix, like `ihello world`

I expected this to happen:

On startup, helix is visibly in insert mode with "hello world" in an empty buffer.

Instead, this happened:

The input appears to have no affect, but as soon as the first keypress is entered after startup, it becomes clear that the prior input did have an effect, only it was not displayed until the most recent keypress.

### Platform

macOS

### Terminal Emulator

kitty 0.26.2 (also happens with the builtin Terminal Version 2.12.7)

### Helix Version

22.08.1-400-g535cf900

**Comment by mtoohey31** [comment]
I have checked, and the same issue is not present on Linux or Windows.

**Comment by mtoohey31** [comment]
Ah, I can reproduce the `ENTER` causing changes to appear immediately too @KittyBorgX. I wonder if that is an indicator that something strange is going on with how stdin is buffered initially?

I know this bug is kind of a nitpick, but I do run into this pretty frequently, since I'm often closing and re-opening files, and I know what keypresses I need to enter before helix has started. I'm not sure that pressing `ENTER` is really a fix, since that's not usually what I want to input. But again, it's a small issue, so if maintainers don't think it's important, I'm ok with closing this.

**Comment by mtoohey31** [comment]
Also, just to be clear @KittyBorgX, do you see the same behaviour as I do if you don't include `ENTER` in your initial input?

**Comment by KittyBorgX** [comment]
Yup, I can confirm it doesn't work when `ENTER` isn't pressed. It works without the enter key in neovim.

**Comment by KittyBorgX** [comment]
@mtoohey31 Upon further investigation, 
The text does appear after a key is pressed when the scratch buffer opens. 

Here's a demo: 
https://user-images.githubusercontent.com/82328083/200729080-e33ee984-b00a-4a12-8490-f03db4c782bc.mov

I'm unsure of why this happens but it's a bug.

**Comment by mtoohey31** [comment]
Thanks for uploading the demo @KittyBorgX! That's the behaviour that I was trying to explain in the original issue. I've updated the description of the issue to mention your findings related to the enter key, and I also added some guesses about why that might be happening. I haven't had time to dive into the code yet though.

**Comment by KittyBorgX** [comment]
Taking into consideration that this can't be reproduced on Linux or Windows, I think we can conclude that this is a problem of how macOS handles keypresses rather than helix.  (Though I could very well be wrong)

**Comment by KittyBorgX** [comment]
Taking into consideration that this can't be reproduced on Linux or Windows, I think we can conclude that this is a problem of how macOS handles keypresses rather than helix.  (Though I could very well be wrong)

**Comment by CalebLarsen** [comment]
I cannot reproduce this on:
macOS 15.5
kitty 0.37.0
helix 25.01.1 (3ceae88c)

Can anyone confirm if this is still a problem for them?

**Comment by mtoohey31** [comment]
I no longer use macos or kitty (though I do still use helix 🙂) so I can't check. If it doesn't seem to be a problem anymore then I don't have any objections to closing it.

**Comment by forivall** [comment]
I can still reproduce this on iTerm2 on macos, with a version built from `master` (`helix 25.07.1 (ce351f4b)`). With the version installed from homebrew, or built from the git tag (`helix 25.07.1 (a05c151b)`), this issue doesn't occur. So it seems like this was fixed, but has since regressed on `master`.

From a `git bisect` this regressed from 83abbe56dfaf094ffb5b053b49c7daa10b393cfb "Replace Crossterm with Termina"
